### PR TITLE
Add p2 network

### PR DESCRIPTION
**Pull request by unDemian**
**Built on top of https://github.com/Automattic/wp-calypso/pull/42155**

#### Changes proposed in this Pull Request

- Team section was renamed to A8C since it was used for A8C Followed
- Add a P2 Network Reader section which should include user's wp4teams sites.
Markup 2020-05-12 at 9 44 31 PM

#### Testing instructions
- Apply D43646-code
- Apply D43648-code
- Apply D43649-code
- Apply D43808-code
- Test Reader P2 Network unseen status and mark as seen/unseen work as expected

### CODE REVIEW

**Comment by evilluendas** [comment]
Nice work!

One thing I've noticed is that it displays _all_ the P2 sites you're part of, right? I see a ton of P2s in that list that I don't really have any particular interest in, but I joined at some point because internal A8C P2s incentivize you to do so (for example, to be able to see them on the phone without being proxied) even if they don't involve you directly. 

I think it would make sense to only show the P2 sites that you actually follow, so you don't have all that noise and can focus on the sites that you're really interested in. It's how the Automattic tab has always worked, and it makes sense in the context of the Reader: I think should only show you stuff that you want to follow (aside from specific sections like Discover that do the opposite).

So, to summarize, in my opinion:
- The P2 Network "folder" should only show the **non-A8C** P2s that you follow. If you don't follow any, the folder is not shown. Since creating or joining a P2 site should make you auto-follow it, it'd appear here.
- I would rename it from "P2 Network" to just "P2". We can eventually swap the globe icon with the P2 logo too.
- We should have a separate folder called "Automattic" that works in the same way but only contains the A8C P2s that you follow.
- Ideally, we should add per-site counters at this point, so you have all the sites you follow _and_ see which ones have new content.
- After this we can eventually remove the old "Automattic" tab (which only shows a single feed of all the sites that you follow combined) because it'd be obsolete with this new per-site approach.
- Once we have multi-site, different networks should behave the same way: each one with their own folder and the followed sites inside, etc.

I have some mockups halfway done, so I'll finish it and post it to our P2 for posterity!

**Comment by unDemian** [comment]
Thanks for taking the time, feedback addressed.

> I think it would make sense to only show the P2 sites that you actually follow, so you don't have all that noise and can focus on the sites that you're really interested in. It's how the Automattic tab has always worked, and it makes sense in the context of the Reader: I think should only show you stuff that you want to follow (aside from specific sections like Discover that do the opposite).

If you pull latest and apply D43808-code also, the P2 section would only show non-a8c p2s.

> We should have a separate folder called "Automattic" that works in the same way but only contains the A8C P2s that you follow.

I'll have it in a different PR.

> Ideally, we should add per-site counters at this point, so you have all the sites you follow and see which ones have new content.

Will be added in future PRs

File: client/reader/following/main.jsx [hunk 1]
```diff
@@ -96,12 +96,12 @@ const FollowingStream = ( props ) => {
 			<SectionHeader label={ translate( 'Followed Sites' ) }>
 				{ config.isEnabled( 'reader/seen-posts' ) && ! props.hasUnseen && (
 					<Button compact onClick={ markAllAsUnSeen }>
-						{ translate( 'Mark All as Unseen' ) }
+						{ translate( 'Mark All as unseen' ) }
```
Comment (a8ci18n) [inline client/reader/following/main.jsx:99]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/reader/following/main.jsx [hunk 2]
```diff
@@ -96,12 +96,12 @@ const FollowingStream = ( props ) => {
...
 					</Button>
 				) }
 				{ config.isEnabled( 'reader/seen-posts' ) && props.hasUnseen && (
 					<Button compact onClick={ markAllAsSeen }>
-						{ translate( 'Mark All as Seen' ) }
+						{ translate( 'Mark All as seen' ) }
```
Comment (a8ci18n) [inline client/reader/following/main.jsx:104]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Mark all as seen' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/unseen-for-subsections)

File: client/reader/sidebar/reader-sidebar-network/index.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	render() {
+		const { isOpen, translate, onClick, unseen } = this.props;
+		return (
+			<ul className={ unseen && unseen.status ? 'has-unseen' : '' }>
+				{ /* eslint-disable-next-line wpcalypso/jsx-classname-namespace */ }
+				{ unseen && unseen.status && <span className="sidebar__bubble" /> }
+				<ExpandableSidebarMenu
+					expanded={ isOpen }
+					title={ translate( 'P2' ) }
```
Comment (a8ci18n) [inline client/reader/sidebar/reader-sidebar-network/index.jsx:38]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **6** times:
`translate( 'P2s' )` *ES Score: 7*

**Comment by lezama** [comment]
On first try using the calypso.live branch I got
```async-load-reader-sidebar.36edf5b27f1f2f5728b0.min.js:1 Uncaught (in promise) TypeError: Cannot read property 'ID' of null```

On refreshing it loads without errors

**Comment by lezama** [comment]
Other than the error above it seems to work great.

File: client/reader/a8c/main.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	const markAllAsUnSeen = () => {
+		dispatch( requestMarkAllAsUnseen( { section: SECTION_A8C_FOLLOWING } ) );
+	};
+
+	return (
+		<Stream { ...props } shouldCombineCards={ false }>
+			<SectionHeader label={ translate( 'Followed A8C Sites' ) }>
+				{ config.isEnabled( 'reader/seen-posts' ) && ! props.hasUnseen && (
+					<Button compact onClick={ markAllAsUnSeen }>
+						{ translate( 'Mark All as unseen' ) }
```
Comment (a8ci18n) [inline client/reader/a8c/main.jsx:36]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Mark All as Unseen' )` *ES Score: 14*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/unseen-for-subsections)

File: client/reader/a8c/main.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+					</Button>
+				) }
+				{ config.isEnabled( 'reader/seen-posts' ) && props.hasUnseen && (
+					<Button compact onClick={ markAllAsSeen }>
+						{ translate( 'Mark All as seen' ) }
```
Comment (a8ci18n) [inline client/reader/a8c/main.jsx:41]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Mark all as seen' )` *ES Score: 12*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/unseen-for-subsections)

**Comment by unDemian** [comment]
> Other than the error above it seems to work great.

I could not replicate it, might have been some caching issue.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3734131

Thank you @unDemian for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3734131).

**Pull request merged**